### PR TITLE
docs: Adds coder_git_auth deprication notice (#203)

### PR DESCRIPTION
**Pull request by kyle-pazdel**
Adds deprication notice to git_auth in docs and renames 'gh' to 'GitHub' in external_auth docs for clarity.

### CODE REVIEW

**Comment by kyle-pazdel** [comment]
I have read the CLA Document and I hereby sign the CLA
recheck

**Comment by kyle-pazdel** [comment]
recheck

File: docs/data-sources/external_auth.md [hunk 1]
```diff
@@ -3,16 +3,15 @@
 page_title: "coder_external_auth Data Source - terraform-provider-coder"
 subcategory: ""
 description: |-
-  Use this data source to require users to authenticate with an external service prior to workspace creation. This can be used to pre-authenticate external services in a workspace. (e.g. gcloud, gh, docker, etc)
```
Comment (matifali) [inline docs/data-sources/external_auth.md:6]: `gcloud`, `gh` and `docker` are the binary names. IMO, `gh` makes better sense.

**Review by matifali — CHANGES_REQUESTED** [review]
docs are generated files so we should update the description in the go code,
for example,
https://github.com/coder/terraform-provider-coder/blob/8c5e8ff9579d1d766d5317422bf3c1988b710434/provider/gitauth.go#L16

**Comment by matifali** [comment]
Can be closed if we do #250.

**Pull request closed without merging**